torch/csrc/jit/pickle.h

```diff
@@ -0,0 +1,53 @@
+#include <torch/csrc/WindowsTorchApiMacro.h>
+#include <ATen/core/ivalue.h>
+#include <torch/csrc/jit/pickler.h>
+
+
+namespace torch {
+namespace jit {
+
+/// Pickled values can be loaded in Python and C++:
+///
+/// \rst
+/// .. code-block:: cpp
+///
+///   torch::IValue float_value(2.3);
+///   std::string data = torch::jit::pickle({float_value});
+///   std::ofstream out("data.pkl");
+///   out << data;
+///   out.flush();
+///
+///
+///   std::ifstream in("data.pkl", std::ios::binary);
+///   std::vector<torch::IValue> ivalues = torch::jit::unpickle(in);
+///   std::cout << ivalues.at(0) << "\n";
+///
+/// .. code-block:: python
+///
+///   values = torch.load('data.pkl')
+///   print(values)
+///
+/// \endrst
+TORCH_API std::string pickle(
```

Comment:
If you look at the python pickling apis, they do not force you to take and return vectors, and we have ivalue lists. So I think the API should mirror the python ones, i.e. It should take a single IValue and return a single ivalue. This mirrors how function returns work in TorchScript as well.